### PR TITLE
chore: bump codegen dependency versions to latest tag

### PR DESCRIPTION
**Pull request by phani-srikar**
## Problem

We are bumping the dependency on the codegen packages to the latest tag available. Click here to see the [changeset](https://github.com/aws-amplify/amplify-codegen/compare/amplify-codegen%404.9.0-gen2-release-0411.0...amplify-codegen%404.9.0-gen2-release-0416.0).

Please note that some of the package-lock updates that remove the `beta` resolutions for packages in this repo are probably desirable since these packages are actually out of `beta` on latest `main`.

**Issue number, if available:**

## Changes

**Corresponding docs PR, if applicable:**

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: packages/client-config/API.md [hunk 1]
```diff
@@ -208,7 +208,7 @@ export type CustomClientConfig = {
 export const DEFAULT_CLIENT_CONFIG_VERSION: ClientConfigVersion;
 
 // @public
-export const generateClientConfig: <T extends "1" | "0">(backendIdentifier: DeployedBackendIdentifier, version: T, awsClientProvider?: AWSClientProvider<{
+export const generateClientConfig: <T extends "0" | "1">(backendIdentifier: DeployedBackendIdentifier, version: T, awsClientProvider?: AWSClientProvider<{
```
Comment (sobolk) [inline packages/client-config/API.md:211]: PR checks might complain about this. If they do please revert this change.
(for some reason its being non deterministic between platforms/OSes).

**Review by sobolk — APPROVED** [review]

**Review by palpatim — DISMISSED** [review]

File: packages/client-config/API.md (hunk 1 above)
Comment (phani-srikar) [reply packages/client-config/API.md]: Reverted this file

**Review by sobolk — APPROVED** [review]
